### PR TITLE
[Weighted Routing] Add support for discovered master and remove local weights in the response

### PR DESCRIPTION
**Pull request by anshu1106**
Signed-off-by: Anshu Agarwal <anshukag@amazon.com>

### Description
The local weights don't add much value since weights are relative and having just the local value gives no indication of how traffic would get distributed.
Discovered master is helpful to understand cases where after weights are set and also that the node is unable to connect to the leader either due to
Master not discovered -- This might happen to newly launched after decommission or even otherwise
Decommissioned -- The node was decommissioned

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [ ] New functionality has been documented.
  - [x] New functionality has javadoc added
- [x] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by gbbafna** [comment]
I think we should let this functionality be as is . A local collector could use this to find out weights in a light weight manner , which otherwise  would be  heavy to go to the cluster manager.

**Comment by Bukhtawar** [comment]
> I think we should let this functionality be as is . A local collector could use this to find out weights in a light weight manner , which otherwise would be heavy to go to the cluster manager.

@anshu1106 can correct me but this doesn't remove `_local` query but removes local_weight since it doesn't add much value as weights are relative. The caller is responsible to select specific weights based on the awareness attribute

**Comment by gbbafna** [comment]
> > I think we should let this functionality be as is . A local collector could use this to find out weights in a light weight manner , which otherwise would be heavy to go to the cluster manager.
> 
> @anshu1106 can correct me but this doesn't remove `_local` query but removes local_weight since it doesn't add much value as weights are relative. The caller is responsible to select specific weights based on the awareness attribute

Got it. My bad, i misread the title  .

File: server/src/internalClusterTest/java/org/opensearch/cluster/routing/WeightedRoutingIT.java [hunk 1]
```diff
@@ -219,7 +219,6 @@ public void testGetWeightedRouting_WeightsAreSet() throws IOException {
             .setRequestLocal(true)
             .get();
         assertEquals(weightedRouting, weightedRoutingResponse.weights());
-        assertEquals("1.0", weightedRoutingResponse.getLocalNodeWeight());
```
Comment (gbbafna) [inline server/src/internalClusterTest/java/org/opensearch/cluster/routing/WeightedRoutingIT.java:222]: add `discovered_cluster_manager` assertion as well .

File: server/src/main/java/org/opensearch/action/admin/cluster/shards/routing/weighted/get/ClusterGetWeightedRoutingResponse.java [hunk 2]
```diff
@@ -31,26 +31,37 @@
...
-    private static final String NODE_WEIGHT = "node_weight";
+    public WeightedRouting getWeightedRouting() {
+        return weightedRouting;
+    }
+
+    private final WeightedRouting weightedRouting;
 
-    public String getLocalNodeWeight() {
-        return localNodeWeight;
+    public Boolean getDiscoveredMaster() {
```
Comment (gbbafna) [inline server/src/main/java/org/opensearch/action/admin/cluster/shards/routing/weighted/get/ClusterGetWeightedRoutingResponse.java:40]: lets use `cluster_manager` everywhere . `master` is going to get deprecated in 3.0.

**Review by gbbafna — CHANGES_REQUESTED** [review]

**Review by gbbafna — APPROVED** [review]

File: server/src/main/java/org/opensearch/action/admin/cluster/shards/routing/weighted/get/ClusterGetWeightedRoutingResponse.java [hunk 3]
```diff
@@ -31,26 +31,37 @@
...
     }
 
+    private final Boolean discoveredClusterManager;
+
+    private static final String DISCOVERED_CLUSTER_MANAGER = "discovered_cluster_manager";
+
     ClusterGetWeightedRoutingResponse() {
         this.weightedRouting = null;
+        this.discoveredClusterManager = null;
     }
```
Comment (Bukhtawar) [inline server/src/main/java/org/opensearch/action/admin/cluster/shards/routing/weighted/get/ClusterGetWeightedRoutingResponse.java:51]: There is no explicit need to set them null

**Review by Bukhtawar — APPROVED** [review]

**Pull request merged**